Sources/PT.PM.Matching/Patterns/PatternAny.cs

```diff
@@ -1,19 +1,54 @@
-﻿using PT.PM.Common.Nodes;
+﻿using System.Text.RegularExpressions;
+using PT.PM.Common.Nodes;
+using PT.PM.Common.Nodes.Tokens;
 
 namespace PT.PM.Matching.Patterns
 {
-    public class PatternAny : PatternUst<Ust>
+    public class PatternAny : PatternUst<Ust>, IRegexPattern
     {
         public static PatternAny Instance = new PatternAny();
 
+        public string Default => ".*";
+
+        public string RegexString
+        {
+            get => Regex.ToString();
+            set => Regex = new Regex(string.IsNullOrEmpty(value) ? Default : value, RegexOptions.Compiled);
+        }
+
+        public Regex Regex { get; private set; }
+
+        public override bool Any => Regex == null || RegexString.Equals(Default);
+
+        public PatternAny() { }
+
+        public PatternAny(string regex)
+        {
+            RegexString = regex;
+        }
+
         public override MatchContext Match(Ust ust, MatchContext context)
         {
-            if (ust == null)
+            if (Any)
             {
-                return context.MakeSuccess();
+                if (ust == null)
+                {
+                    return context.MakeSuccess();
+                }
+                return context.AddMatch(ust);
             }
+            string ustString = ust.ToString();
+            var match = Regex.Match(ustString);
+            if (match.Value.Equals(ustString))
+            {
+                return context.AddMatch(ust);
+            }
+            return context.Fail();
```

Comment:
This pattern can match several text spans for one UST node. Use [MatchRegex](https://github.com/PositiveTechnologies/PT.PM/blob/dev/Sources/PT.PM.Matching/PatternUtils.cs#L10) method from PatternUtils.

See example of using in: [PatternStringRegexLiteral](https://github.com/PositiveTechnologies/PT.PM/blob/dev/Sources/PT.PM.Matching/Patterns/PatternStringRegexLiteral.cs#L38).

Reply:
now a part of logic locate at TextUtils